backend-rust/src/server/rest/rooms_rest_controller.rs

```diff
@@ -0,0 +1,18 @@
+use crate::server::dto::room::RoomDTO;
+use crate::server::error::ServerError;
+use crate::server::state::AppState;
+use axum::extract::State;
+use std::sync::Arc;
+
+pub async fn list_rooms_handler(
+    State(state): State<Arc<AppState>>,
+) -> Result<axum::Json<Vec<RoomDTO>>, ServerError> {
+    let room_data_list = state.room_storage.list_rooms().await?;
+
+    let room_dto_list: Vec<RoomDTO> = room_data_list
```

Comment:
Eu tô chutando que o Rust vai conseguir inferir o tipo dessa variável porque o retorno da função pede `Vec<RoomDTO>`

```suggestion
    let room_dto_list = room_data_list
```